js/views/key_verification_view.js

```diff
@@ -12,59 +12,80 @@
               .then(this.render.bind(this));
               //.then(this.makeQRCode.bind(this));
         },
         makeQRCode: function() {
             // Per Lilia: We can't turn this on until it generates a Latin1 string, as is
             //   required by the mobile clients.
             new QRCode(this.$('.qr')[0]).makeCode(
-                dcodeIO.ByteBuffer.wrap(this.our_key).toString('base64')
+                dcodeIO.ByteBuffer.wrap(this.ourKey).toString('base64')
             );
         },
         loadTheirKey: function() {
-            if (this.their_key) {
-                return Promise.resolve(this.their_key);
-            } else {
-                return textsecure.storage.protocol.loadIdentityKey(
-                    this.model.id
-                ).then(function(their_key) {
-                    this.their_key = their_key;
-                }.bind(this));
-            }
+            return textsecure.storage.protocol.loadIdentityKey(
+                this.model.id
+            ).then(function(theirKey) {
+                this.theirKey = theirKey;
+            }.bind(this));
         },
         loadOurKey: function() {
-            if (this.our_key) {
-                return Promise.resolve(this.our_key);
-            } else {
-                return textsecure.storage.protocol.loadIdentityKey(
-                    this.our_number
-                ).then(function(our_key) {
-                    this.our_key = our_key;
-                }.bind(this));
-            }
+            return textsecure.storage.protocol.loadIdentityKey(
+                this.ourNumber
+            ).then(function(ourKey) {
+                this.ourKey = ourKey;
+            }.bind(this));
         },
         generateSecurityNumber: function() {
             return new libsignal.FingerprintGenerator(5200).createFor(
-                this.our_number, this.our_key, this.model.id, this.their_key
+                this.ourNumber, this.ourKey, this.model.id, this.theirKey
             ).then(function(securityNumber) {
                 this.securityNumber = securityNumber;
             }.bind(this));
         },
+        onSafetyNumberChanged: function() {
+            this.model.getProfiles().then(this.loadKeys.bind(this));
+
+            var dialog = new Whisper.ConfirmationDialogView({
+                message: i18n('changedRightAfterVerify', this.model.getTitle(), this.model.getTitle()),
+                resolve: function() {},
+            });
+
+            dialog.$el.insertBefore(this.el);
+            dialog.focusCancel();
+        },
         toggleVerified: function() {
-            this.model.toggleVerified();
+            this.$('button.verify').attr('disabled', true);
+            this.model.toggleVerified().catch(function(result) {
+                if (result instanceof Error) {
+                    if (result.name === 'OutgoingIdentityKeyError') {
+                        this.onSafetyNumberChanged();
+                    }
```

Comment:
`else { console.log(result); }`, so we can debug all the other exciting ways we might fail!